### DISCUSSION TITLE
Repeater with Unique Values

### DISCUSSION THREAD

**Discussion by st11x** [directus/directus#12263] [Archive]
Hi,

I created a Repeater field and checked the Value has to be unique checkbox but it doesn't do what I think it is supposed to do. My repeated fields are just simple strings.

This is my field.

![image](https://user-images.githubusercontent.com/7398499/159194179-f88f100f-149c-4150-848b-c6619885065c.png)

And in the item, I was still able to create 2 fields with the same string

![image](https://user-images.githubusercontent.com/7398499/159194224-590af341-5163-4c9b-a968-86fa8bc217c5.png)

Did I misunderstand the purpose of the checkbox? If I did, how can I avoid duplicate values? 

thanks
  Matt

**Comment by azrikahar** [comment]
As repeaters are essentially a json field, the `unique` here does actually mean "unique" compared to the repeater in another item/row, not the entries within the same repeater itself, if that makes sense.

I believe there are 2 alternatives you can opt for:

- Use a relational field like M2O so each entry are actual rows in the database which should prompt "value is unique" error when creating new ones.

- Use a [custom filter hook](https://docs.directus.io/extensions/hooks/#filter) which then checks for uniqueness within the repeater field and throw an error when the entries are not unique. You can refer here for code snippets of slightly similar use case: https://github.com/directus/directus/discussions/11504#discussioncomment-2132156

**Reply by st11x** [reply]
Thank you. I was trying to avoid a relationship field so I’ll look into using a custom filter hook.

**Reply by enzolupia** [reply]
@azrikahar  is it possible to use relational fields inside a standard (non custom interface) repeater?

**Reply by azrikahar** [reply]
@enzolupia it is not possible as of now, but there are discussions on this such as #11181, #3962 and #3884.
